Signal/src/textsecure/Messages/TSMessagesManager.m

```diff
@@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
 }
 
 - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
+    if(content.group!= nil )  {
+        __block BOOL ignoreMessage = NO;
+        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
+            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
+            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
+            if(gThread==nil) {
+                ignoreMessage = YES;
+            }
+        }];
+        if(ignoreMessage) {
+            DDLogDebug(@"recevied message from group that I left, ignoring");
```

Comment:
@corbett I would strongly advocate against that server-side option.

The server IS and SHOULD stay dumb. If we're letting the server know in any way that you left a group, that would be information leakage. Those protocol buffers are encrypted and the server doesn't know it's a group message. So that is not even technically feasible.